components/resourceDetails/resourceSharing/addAgentButton/agentPickerModal/index.test.jsx

```diff
@@ -0,0 +1,348 @@
+        />
+      </AccessControlContext.Provider>
+    );
+
+    const addWebIdButton = getByTestId("add-webid-button");
+    userEvent.click(addWebIdButton);
+    const submitWebIdsButton = getByTestId("submit-webids-button");
+    userEvent.click(submitWebIdsButton);
+
+    expect(getByRole("alert")).not.toBeNull();
+  });
+
+  it("opens a confirmation dialog", async () => {
+    const setOpen = jest.fn();
+    const setTitle = jest.fn();
+
+    const contextValue = {
+      confirmed: false,
+      content: null,
+      open: false,
+      setConfirmed: jest.fn(),
+      setContent: jest.fn(),
+      setOpen,
+      setTitle,
+      title: "Confirmation",
+    };
+    const { getByTestId, findByText, findByTestId } = renderWithTheme(
+      <ConfirmationDialogContext.Provider value={contextValue}>
+        <AccessControlContext.Provider value={{ accessControl }}>
+          <AgentPickerModal
+            type="editors"
+            text="Add Editors"
+            onClose={onClose}
+            mutatePermissions={mutatePermissions}
+            permissions={permissions}
+          />
+        </AccessControlContext.Provider>
+      </ConfirmationDialogContext.Provider>
+    );
+
+    const webId = "https://somewebid.com";
+
+    await fetchProfile.mockRejectedValue({ error: "error" });
+
+    const addWebIdButton = getByTestId("add-webid-button");
+    userEvent.click(addWebIdButton);
+    const input = await findByTestId("webid-input");
+    userEvent.type(input, webId);
+    const addButton = getByTestId("add-button");
+    userEvent.click(addButton);
+    await findByText(webId);
+    const submitWebIdsButton = getByTestId("submit-webids-button");
+    userEvent.click(submitWebIdsButton);
+    expect(setTitle).toHaveBeenCalledWith("Change permissions for 1 person");
+    expect(setOpen).toHaveBeenCalledWith("add-new-permissions");
+  });
+  it("renders a warning when trying to submit a webId that is already in the policy", async () => {
+    const webId = "https://somewebid.com";
+
+    const { getByTestId, findByText, findByTestId } = renderWithTheme(
+      <AccessControlContext.Provider value={{ accessControl }}>
+        <AgentPickerModal
+          type="editors"
+          text="Add Editors"
+          onClose={onClose}
+          mutatePermissions={mutatePermissions}
+          permissions={[{ webId }]}
+        />
+      </AccessControlContext.Provider>
+    );
+
+    const addWebIdButton = getByTestId("add-webid-button");
+    userEvent.click(addWebIdButton);
+    const input = await findByTestId("webid-input");
+    userEvent.type(input, webId);
+    const addButton = getByTestId("add-button");
+    userEvent.click(addButton);
+    const submitWebIdsButton = getByTestId("submit-webids-button");
+    userEvent.click(submitWebIdsButton);
+    await waitFor(() => {
+      expect(findByText("That WebID has already been added")).not.toBeNull();
+    });
+  });
+});
+describe("handleConfirmation", () => {
```

Comment:
Copying comment from above:

So, these are the components (this and the remove button) where I was forced to do this because I cannot test the interactions with the Confirmation Dialog. I don't have any way of testing this without externalizing them, so I had no choice. I would like, however, to refactor the confirmation dialog and revisit these two components' tests when that is done.